src/Microsoft.IdentityModel.Tokens/JsonWebKeyConverter.cs

```diff
@@ -336,6 +368,37 @@ internal static bool TryCreateToRsaSecurityKey(JsonWebKey webKey, out SecurityKe
             return false;
         }
 
+        internal static bool TryCreateToRsaSecurityKey(JsonWebKey webKey, IDictionary<JsonWebKey, List<string>> convertKeyInfos, out SecurityKey key)
+        {
+            if (webKey.ConvertedSecurityKey is RsaSecurityKey)
+            {
+                key = webKey.ConvertedSecurityKey;
+                return true;
+            }
+
+            key = null;
+            if (string.IsNullOrWhiteSpace(webKey.E) || string.IsNullOrWhiteSpace(webKey.N))
+                return false;
+
+            try
+            {
+                key = new RsaSecurityKey(webKey);
+                return true;
+            }
+            catch (Exception ex)
+            {
+                string convertKeyInfo = LogHelper.FormatInvariant(LogMessages.IDX10813, LogHelper.MarkAsNonPII(typeof(RsaSecurityKey)), webKey, ex);
+                if (convertKeyInfos.ContainsKey(webKey))
+                    convertKeyInfos[webKey].Add(convertKeyInfo);
+                else
+                    convertKeyInfos.Add(webKey, new List<string> { convertKeyInfo });
+
+                LogHelper.LogExceptionMessage(new InvalidOperationException(convertKeyInfo, ex));
+            }
```

Comment:
For at least this case we definitely should not use this patter to achieve the TryCreate pattern.

We control the RsaSecurityKey type. There's no reason for us to use exception handling to perform this check, performance isn't good plus it's a place for bugs to hide.

I'd recommend refactoring RsaSecurityKey to initially call an internal validate method which can return an exception. In the Init method this can just be thrown, and in this method anything returned implies we should return false.

I suspect we can avoid the bad "catch all exceptions pattern" in the other methods too.

IMO "Used as validation" is not sufficient justification to suppress CA1031